app/services/gobierto_common/context_service.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module GobiertoCommon
+  class ContextService
+    attr_reader :context
+
+    delegate :present?, to: :resource
+
+    def initialize(context)
+      @context = fill_missing_parts(context.to_s.strip)
+    end
+
+    def resource
+      @resource ||= begin
+                      GlobalID::Locator.locate @context
+                    rescue ActiveRecord::RecordNotFound
+                      nil
+                    end
+    end
+
+    private
+
+    def fill_missing_parts(context)
+      return context if context =~ %r{^gid://gobierto//}
+
+      missing_part = "gid://"
```

Comment:
You should better use `URI::GID` API, i.e `URI::GID.create('app', model).to_s` https://github.com/rails/globalid/blob/master/lib/global_id/uri/gid.rb

Reply:
In this case the service defines this method to add missing parts to the gid full URI string if required, but the URI::GID class doesn't seem to provide a method to do this, and it doesn't also provide tools to manage a global id as a string it provides a mehtod to create a global id from fragments so I'd have to split a GID string previously.